### DISCUSSION TITLE
Increasing the Number of Wind Measurement Points

### DISCUSSION THREAD

**Discussion by venturi123** [OpenFAST/openfast#2316] [Q&A]
Hello everyone,

I've encountered an issue with the FAST.Farm, which currently supports wind speed measurements at only up to 9 points in the flow field. However, for my project, I need to measure wind speed at more than 9 points.

Could anyone please guide me on how to overcome this limitation? Are there any workarounds or modifications that can be made to the input files or the software itself to allow for more measurement points?

Any suggestions or solutions would be greatly appreciated!

Thank you in advance for your help.

Best regards,

**Comment by jjonkman** [comment]
Dear @venturi123,

Certainly internal to FAST.Farm, FAST.Farm knows the disturbed wind at all points in the domain, not just at 9 points.  So, you could always change the source code to increase `NWindVel` from max 9 to something higher, but this would require that you change the wind-related outputs in the `OutList` (`WηVDisX`, etc.) to use two (for up to 99 points) or more digits to define point η.

Alternatively, instead of using the `OutList` to output wind data, you can output slices of the disturbed wind domain via the VISUALIZATION-related outputs already available in FAST.Farm.

Best regards,

**Comment by venturi123** [comment]
Dear @jjonkman ,

I greatly appreciate your suggestions. However, I am concerned that the I/O operations for VTK files might become a bottleneck in my simulations, as I need to simulate a series of parametric cases. In addition, I have tried revising the source code for my purposes, but it does not seem to be straightforward.

Hence, I investigated the LIDAR feature introduced in version 3.5.x for indirectly sensing wind speeds, but I encountered the same issue described in https://github.com/OpenFAST/openfast/discussions/2324#discussion-6936979.

Here are the errors raised in the terminal:
```
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source             
libc.so.6          0000751405642520  Unknown               Unknown  Unknown
FAST.Farm          0000000001C64784  Unknown               Unknown  Unknown
FAST.Farm          0000000001C4C710  Unknown               Unknown  Unknown
FAST.Farm          00000000005C277A  Unknown               Unknown  Unknown
FAST.Farm          00000000005D21F9  Unknown               Unknown  Unknown
FAST.Farm          0000000000431525  Unknown               Unknown  Unknown
FAST.Farm          0000000000450871  Unknown               Unknown  Unknown
FAST.Farm          0000000000407EED  Unknown               Unknown  Unknown
libc.so.6          0000751405629D90  Unknown               Unknown  Unknown
libc.so.6          0000751405629E40  __libc_start_main     Unknown  Unknown
FAST.Farm          0000000000407E05  Unknown               Unknown  Unknown

```

I am currently using FAST.Farm v3.5.3, and here are my configuration settings for the LIDAR parameters. I have already tried many other parameter combinations, but I still encounter the same error:

```
================== LIDAR Parameters ===========================================================================
          3   SensorType          - Switch for lidar configuration (0 = None, 1 = Single Point Beam(s), 2 = Continuous, 3 = Pulsed)
          2   NumPulseGate        - Number of lidar measurement gates (used when SensorType = 3)
         30   PulseSpacing        - Distance between range gates (m) (used when SensorType = 3)
          1   NumBeam             - Number of lidar measurement beams (0-5)(used when SensorType = 1)
     -200.0   FocalDistanceX      - Focal distance co-ordinates of the lidar beam in the x direction (relative to hub height) (only first coordinate used for SensorType 2 and 3) (m)
          0   FocalDistanceY      - Focal distance co-ordinates of the lidar beam in the y direction (relative to hub height) (only first coordinate used for SensorType 2 and 3) (m)
          0   FocalDistanceZ      - Focal distance co-ordinates of the lidar beam in the z direction (relative to hub height) (only first coordinate used for SensorType 2 and 3) (m)
0.0 0.0 0.0   RotorApexOffsetPos  - Offset of the lidar from hub height (m)
        7.5   URefLid             - Reference average wind speed for the lidar[m/s]
       0.25   MeasurementInterval - Time between each measurement [s]
      False   LidRadialVel        - TRUE => return radial component, FALSE => return 'x' direction estimate
          1   ConsiderHubMotion   - Flag whether to consider the hub motion's impact on Lidar measurements
```

I wonder if there are any further suggestions for my situation? Many thanks!

Best,

**Reply by andrew-platt** [reply]
Dear @venturi123,

Like @jjonkman, I recommend using the VTK outputs for sampling a larger number of points.  This will work very well if all your sampling points are withing a plane, but as you point out, this is a huge amount of data.

In regards to the _LiDAR_ outputs, this unfortunately will not work well at the _FAST.Farm_ level.  The _InflowWind_ instance for a single turbine only has access to a small region of wind in the immediate vicinity of the turbine, and _InflowWind_ instances at the _FAST.Farm_ level don't know about the wake affects (this is computed and stored separately, then passed to each turbine).  We have ideas on how to fix this, but don't have funding at the moment to do so.

So unfortunately I don't have a good solution for you other than the VTK writing at the moment.  

As for the segmentation faults with _LiDAR_, this is clearly a bug that we hope to address soon.

Regards,
Andy

**Reply by andrew-platt** [reply]
After further investigation, it is not possible to use the _LIDAR_ from _InflowWind_ at the _FAST.Farm_ level at this time.  The _LIDAR_ needs some information about the location of the turbine hub from _ElastoDyn_ and information from _ServoDyn_, however this information is not available at the _FAST.Farm_ level since this instance of _InflowWind_ is not associated with a turbine.

Usage of the _LiDAR_ with _InflowWind_ is now disabled in the upcoming 3.5.5 release with a warning.
